### ISSUE TITLE
Exclude Life Tap and Drain Life ranks

### ISSUE THREAD

**Issue by Gogo1951** [valkyrnstudios/RankSentinel#89]
https://github.com/Gogo1951/GogoWatch/issues/23

I didn't validate this at all. Just re-posting for @Anonomit

**Comment by SabreValkyrn** [comment]
That's aggressively dramatic and out of the blue. Looks like referring to #84 where they also didn't say anything for 4 months.

**Comment by Gogo1951** [comment]
I don't remember all the details, but I think we agreed back in the day that we weren't going to touch PVP abilities... they do a lot of stuff that's different from PVE there. 

Meh. Just hadn't signed in to GitHub in a while. Cheers mate, wasn't trying to be dramatic or aggressive.

**Comment by SabreValkyrn** [comment]
> Meh. Just hadn't signed in to GitHub in a while. Cheers mate, wasn't trying to be dramatic or aggressive.

Sorry, not you, both issue posts.

**Comment by Anonomit** [comment]
> aggressively dramatic and out of the blue

Not trying to be aggressive, dramatic, or sudden. I just don't know the correct place to suggest a change for this addon. The Curseforge page links here.

I first mentioned the issue almost a year ago, and it was closed after 6 months with no comment. Then I found #65, which was suggested 6 months earlier than mine and was also closed without comment. There's also the related PR #68, which was closed with the given reason that changes must be made to the google sheet. As far as I can tell, Gogo maintains that sheet, so that's why I posted there with an explanation for why I was doing that. If this is wrong then please direct me where to go to suggest a change to this sheet.

> grossly missing information on rationale and client

I think I provided a clear rationale. I assume that your requirement for excluding a spell rank would be that the rank is common/useful to downrank. I don't know your thought process, so if this isn't enough then please let me know what kind of rationale you expect. Life Tap and Health Funnel are in all classic game versions, but if you don't know that then it would be better to ask instead of silently closing the issue. You could also look into [issue templates](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository) if you require any specific format or information for issues. [Here's an example](https://github.com/Anonomit/LootReserve/tree/main/.github/ISSUE_TEMPLATE) from one of my addons, and [this is what it looks like](https://github.com/Anonomit/LootReserve/issues/new/choose) when opening an issue. I'd be happy to reformat the request if that would help you.

To be clear, the suggestion is to exclude all ranks of Life Tap and Health Funnel. They exist in all classic game versions, and are commonly downranked in all situations: leveling, raiding, pvp, etc.

**Comment by Gogo1951** [comment]
Dude... @Anonomit wait until you find out about form-based issues and project boards. (=

I haven't played for a year, but I remember having a conversation with an old friend from EJ around spells Warlocks should down-rank in PVE. His response was, "None of 'em." Meh, like everything there are nuances, special situations, but broadly speaking I think most of the people who down-rank aren't doing it intentionally, they just haven't updated their bars since leveling.

**Comment by Anonomit** [comment]
Fair enough. It's not like I have actual stats on how many people downrank, so maybe it really is more helpful than harmful. For me though, it's just been a lot of spam (before I filtered it out).

Maybe a better solution would be to check if the person has RankSentinel before sending a whisper. If they have the addon, then they ought to know already that they're not using the top rank. There's currently no way to avoid RankSentinel false positives when you're downranking. So even if people had to get the addon themselves to do it, there should be some way to avoid the spam.

**Comment by Gogo1951** [comment]
I think there was a check to see who should message the person, so the person would only get hit once with the message. But... I forget how it all works, it's been a minute. 

Side note, I can't find any documentation saying, "Downrank Life Tap in PVP." And I know Drain Soul Rank 1 was removed from Wrath.

I know with like Growl and Mend Pet it went off all the time, and people were like, "Oh, guess I should have upgraded Growl, and I could be like, 'Fool, turn Growl off...'" Ha. Dunno, I made a Weak Aura to replace the "Who Taunted?" add-on, but I never really got to test it before I quit.

Soon enough Wrath will go away though. Dun dun dun. I really wish they would just leave 1 Wrath Server kicking for me if I ever wanted to come back. =P

**Comment by Anonomit** [comment]
> I think there was a check to see who should message the person, so the person would only get hit once with the message.

I haven't looked at it at all, but I would guess that this tracking is done privately. If you join an LFG dungeon, you'll get a message from each person with the addon, about each ability that you use downranked.

> Side note, I can't find any documentation saying, "Downrank Life Tap in PVP."

I haven't really done any pvp for a few years. Saving resources isn't specific to pvp. It's considerate to not waste health even when you have a healer around, but it's more important when you don't. I mostly downrank Life Tap in between fights to top off my mana without overflowing it and wasting health.

> I really wish they would just leave 1 Wrath Server kicking 

I wish they'd left a TBC server up, even if it was on PTR or something. There have been times when I wanted to test things in TBC. It'll be the same for Wrath.
